packages/zendesk-adapter/src/change_validators/custom_statuses_enabled.ts

```diff
@@ -60,6 +60,75 @@ const areCustomStatusesEnabled = async (
   return customStatusesEnabled.enabled
 }
 
+const createErrorsForCustomStatusTypes = (
+  changes: readonly Change<ChangeDataType>[]
+): ChangeError[] =>
+  changes
+    .filter(
+      change =>
+        getChangeData(change).elemID.typeName === CUSTOM_STATUS_TYPE_NAME
+    )
+    .map(getChangeData)
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Error',
+      message: 'Custom statuses are not enabled.',
+      detailedMessage:
+        'Cannot deploy custom statuses when they are not enabled in the Zendesk account.',
+    }))
+
+type ChildField = {
+  // eslint-disable-next-line camelcase
+  required_on_statuses: {
+    // eslint-disable-next-line camelcase
+    custom_statuses?: ReferenceExpression[]
+  }
+}
+
+type Condition = {
+  // eslint-disable-next-line camelcase
+  child_fields: ChildField[]
+}
+
+type TicketValue = {
+  // eslint-disable-next-line camelcase
+  agent_conditions?: Condition[]
+  // eslint-disable-next-line camelcase
+  end_user_conditions?: Condition[]
+}
+
+const hasConditionWithCustomStatuses = (conditions: Condition[]): boolean =>
+  conditions.some((condition: Condition) =>
+    condition.child_fields.some(
+      field => (field.required_on_statuses.custom_statuses?.length ?? 0) > 0
+    ))
+
+const isTicketFormWithCustomStatus = (change: Change<ChangeDataType>): boolean => {
+  const data = getChangeData(change)
+  if (data.elemID.typeName !== TICKET_FORM_TYPE_NAME) return false
+  if (!isInstanceElement(data)) return false
+
+  const ticketInstance = data as InstanceElement
+  const ticketValue = ticketInstance.value as TicketValue
+
+  return hasConditionWithCustomStatuses(ticketValue.agent_conditions || [])
+    || hasConditionWithCustomStatuses(ticketValue.end_user_conditions || [])
+}
+
+const createErrorsForTicketFormsWithCustomStatuses = (
+  changes: readonly Change<ChangeDataType>[]
+): ChangeError[] =>
+  changes
+    .filter(isTicketFormWithCustomStatus)
+    .map(getChangeData)
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Warning',
+      message: 'Custom statuses are not enabled.',
+      detailedMessage:
+        'Custom statuses are not enabled which which may cause some statuses to change their context',
```

Comment:
I think we should be more explicit - e.g. something like
```suggestion
      message: 'Deploying ticket form with custom statuses while custom statuses are disabled',
      detailedMessage: 'It seems this ticket form originates from another account that has custom statuses enabled. Since custom statuses are disabled in the target account, this ticket form will be deployed without the custom_statuses fields',
```
@tomermevorach wdyt?

Reply:
Done 👍  (I'll update again if needed)